controllers/osbuild_controller.go

```diff
@@ -47,16 +81,220 @@ type OSBuildReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
 func (r *OSBuildReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx).WithValues("osbuild", req.Name)
+
+	osBuild, err := r.OSBuildRepository.Read(ctx, req.Name, req.Namespace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{Requeue: true, RequeueAfter: RequeueForDuration}, err
+	}
+
+	if osBuild.DeletionTimestamp != nil {
+		// The OSBuild CRs that were created by that OSBuildConfig would be deleted
+		// thanks to setting controller reference for each OSBuild CR
+		return ctrl.Result{}, nil
+	}
+
+	var composeStatus composer.ComposeStatusValue
+	containerBuildStatus := osbuildv1alpha1.OSBuildConditionType("")
+	if osBuild.Status.Conditions != nil {
+		conditionLen := len(osBuild.Status.Conditions)
+		containerBuildStatus = osBuild.Status.Conditions[conditionLen-1].Type
+	}
+
+	if osBuild.Status.ContainerComposeId != EmptyComposeID && containerBuildStatus != ContainerBuildDone {
```

Comment:
I think there's a problem in the `if...else` statement.
First run, there is no Compose ID, so you go into the `else`
Second run, there is a Compose ID, but the Status is not Done so you go into the `if`.
Once the status is Done, you'll go into the `else` and reschedule another compose.
First check if ContainerComposeID is empty, if yes, schedule a build and return. If it's not empty, you can continue to check it's status. If it's not Done, you check and requeue until it is. Once it's done you'll continue to do the same for the ISO build

Reply:
I totally changed it, can you r-review it now?